rules/hmap.bzl

```diff
@@ -86,12 +86,11 @@ def _make_headermap_impl(ctx):
     objc_provider = apple_common.new_objc_provider(
         header = depset([ctx.outputs.headermap]),
     )
-    return struct(
-        files = depset([ctx.outputs.headermap]),
-        providers = [objc_provider],
-        objc = objc_provider,
-        headers = depset([ctx.outputs.headermap]),
-    )
+    cc_info_provider = CcInfo(compilation_context = objc_provider.compilation_context)
```

Comment:
there is a failure on MixedSourceFramework where `...MixedSourceFramework-modulemap/MixedSourceFramework-umbrella.h:16:9: error: 'DoubleQuoteLogger.h' file not found`
Since the other PR has changes on this test case, I will do rebase before merge